diagnostic_feedback/public/js/src/student/student_quiz.js

```diff
@@ -144,15 +145,42 @@ function StudentQuiz(runtime, element, initData) {
 
           if (success && response.student_result) {
             //log event for quiz finish
-            event_data.event_type = 'xblock.diagnostic_feedback.quiz.finish';
+
+            var event_data = {
+              event_type: 'xblock.diagnostic_feedback.quiz.finish',
+              quiz_type: initData.quiz_type,
+              quiz_title: initData.quiz_title,
+              current_question: currentStep
+            };
+
             common.publishEvent(event_data);
             showResult(response.student_result);
           }
 
+          var qEventData = getQuestionEventData(choice.student_choice);
```

Comment:
@jareerahsan Nit: Any specific reason why you're not using the more explicit `questionEventData` as a variable name here?